### ISSUE TITLE
Extending array prototype

### ISSUE THREAD

**Issue by holloway** [ManNguyen/LargestRemainderRound#2]
Hi, thanks for this library.

Could you please provide a version of the library that just exports the function(s)?

Your current approach of extending the array prototype can cause problems https://stackoverflow.com/a/21945724

**Comment by nightpool** [comment]
I agree with the above, I was looking for a library for this and yours seems to be the best, but extending the Array prototype makes the library a non-starter for me due to the effects it might have on the rest of my code. For example, in previous projects I've had libraries go into infinite loops, rendering code crash, and all sorts of other issues when prototypes are extended. Not to mention the performance issues. 

It would be great to have a version of this where i could just call `spread(users, {prop: "myValue"})` and get back the right answer without worrying about prototype conflicts
